doc/source/cluster/config.rst

```diff
@@ -1,80 +1,271 @@
+            :ref:`Node config <cluster-configuration-node-config-type>`
+        :ref:`resources <cluster-configuration-resources>`:
+            :ref:`Resources <cluster-configuration-resources-type>`
+        :ref:`min_workers <cluster-configuration-node-min-workers>`: int
+        :ref:`max_workers <cluster-configuration-node-max-workers>`: int
+        :ref:`worker_setup_commands <cluster-configuration-worker-setup-commands>`:
+            - str
+        :ref:`docker <cluster-configuration-node-docker>`:
+            :ref:`Node Docker <cluster-configuration-node-docker-type>`
+    <node_type_2_name>:
+        ...
+    ...
+
+.. _cluster-configuration-node-config-type:
+
+Node config
+~~~~~~~~~~~
+
+.. tabs::
+    .. group-tab:: AWS
+
+        A YAML object as defined in `the AWS docs <https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html>`_.
+
+    .. group-tab:: Azure
+
+        A YAML object as defined in `the deployment template <https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines>`_ whose resources are defined in `the Azure docs <https://docs.microsoft.com/en-us/azure/templates/>`_.
+
+    .. group-tab:: GCP
+
+        A YAML object as defined in `the GCP docs <https://cloud.google.com/compute/docs/reference/rest/v1/instances>`_.
+
+.. _cluster-configuration-node-docker-type:
+
+Node Docker
+~~~~~~~~~~~
+
+.. parsed-literal::
+
+    :ref:`worker_image <cluster-configuration-image>`: str
+    :ref:`pull_before_run <cluster-configuration-pull-before-run>`: bool
+    :ref:`worker_run_options <cluster-configuration-run-options>`:
+        - str
+
+.. _cluster-configuration-resources-type:
+
+Resources
+~~~~~~~~~
+
+.. parsed-literal::
+
+    :ref:`CPU <cluster-configuration-CPU>`: int
+    :ref:`GPU <cluster-configuration-GPU>`: int
+    <custom_resource1>: int
+    <custom_resource2>: int
+    ...
+
+.. _cluster-configuration-file-mounts-type:
 
-    docker:
-        container_name: "ray_container"
-        image: "rayproject/ray-ml:latest-gpu"
+File mounts
+~~~~~~~~~~~
 
-We provide docker images on `DockerHub <https://hub.docker.com/u/rayproject>`__. The ``rayproject/ray-ml:latest`` image is a quick way to get up and running .
+.. parsed-literal::
+    <path1_on_remote_machine>: str # Path 1 on local machine
+    <path2_on_remote_machine>: str # Path 2 on local machine
+    ...
 
-When the cluster is launched, all of the Ray tasks will be executed completely inside of the container. For GPU support, Ray will automatically select the Nvidia docker runtime if available, and you just need to specify a docker image with the CUDA support (``rayproject/ray-ml:latest-gpu`` and all of our ``-gpu`` images have this).
+Properties
+----------
+
+.. _cluster-configuration-cluster-name:
+
+``cluster_name``
+~~~~~~~~~~~~~~~~
+
+The name of the cluster.
+
+* **Required:** Yes
+* **Importance:** High
+* **Type:** String
+* **Pattern:** **TODO**
+* **Update requires:** Restart
```

Comment:
What this field means?

Reply:
I think Pattern is supposed to indicate constraints on the string -- allowable characters, length, etc.

I think the cluster name is only used as a node tag? So the constraint on the string for a given node provider would be whatever won't break the implementation of `set_node_tags`.

So...for the AWSNodeProvider the cluster name has to be a `v` that doesn't break this ec2 call...
https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/aws/node_provider.py#L234-L240